n6010 event log

hundeboll-silicom opened the pull request:
Example short output:
```
$ sudo fpgainfo events -b3
Boot 3
  Power On Status Time                               : Thu Jan  1 01:23:04 1970
    Status                                           : 0x22000000
      Power On Code FPGA                             : 0x2
      Power On Code CVL                              : 0x2
    FPGA_Status                                      : 0x02ca9029
      FPGA Page                                      : 0x1
      FPGA Configued Page                            : 1
      FPGA Config Timeline                           : 0
      Flow Done                                      : 1
      FSM State                                      : 0x9
    FPGA_Config Status                               : 0x0007e412
      Config Status                                  : 0x2
      Config FW Seq Fail                             : 1
      nStatus Stuck Low Restarts                     : 0x0
      Image Fail Recycle Count                       : 0x0
      1st Power On Page number                       : 0x1
      2nd Power On Page number                       : 0x7
      3nd Power On Page number                       : 0x7
  Power Off Status Time                              : Thu Jan  1 02:20:04 1970
    Power Good Status                                : 0x00000000
    Power Good Record 1                              : 0x80000000
    Power Good Record 2                              : 0x00000000
    Sequencer Status 1                               : 0x80001000
    Sequencer Status 2                               : 0x00914000
    GPI Status                                       : 0x000000bc
    Sensor Failed                                    : 0x000013ff
    Sensor Alert 1                                   : 0x00000000
    Sensor Alert 2                                   : 0x00000000
    Sensor Alert 3                                   : 0x00000000
  PCI Error Status Time                              : Thu Jan  1 02:20:14 1970
```

Example long output:
```
$ sudo fpgainfo events -b3 -s -i
Boot 3
  Power On Status Time                               : Thu Jan  1 01:23:04 1970
    Status                                           : 0x22000000
      Power On Code FPGA                             : 0x2
      Power On Code CVL                              : 0x2
    FPGA_Status                                      : 0x02ca9029
      FPGA Page                                      : 0x1
      FPGA Configued Page                            : 1
      FPGA Config Timeline                           : 0
      Flow Done                                      : 1
      FSM State                                      : 0x9
    FPGA_Config Status                               : 0x0007e412
      Config Status                                  : 0x2
      Config FW Seq Fail                             : 1
      nStatus Stuck Low Restarts                     : 0x0
      Image Fail Recycle Count                       : 0x0
      1st Power On Page number                       : 0x1
      2nd Power On Page number                       : 0x7
      3nd Power On Page number                       : 0x7
  Power Off Status Time                              : Thu Jan  1 02:20:04 1970
    Power Good Status                                : 0x00000000
      pg_12v_aux_efuse                               : 0
      pg_12v_pcie_efuse                              : 0
      pg_vcc_5v                                      : 0
      pg_vcc_3v3                                     : 0
      pg_vccl_fpga_vid                               : 0
      pg_vccl_sdm_0v8                                : 0
      pg_vcch_0v9                                    : 0
      pg_vcch_gxer_1v1                               : 0
      pg_vcca_1v8                                    : 0
      pg_vccclk_gxer_2v5                             : 0
      pg_vpp_2v5                                     : 0
      pg_vccio_1v8                                   : 0
      pg_sdm_1v8                                     : 0
      pg_vcc_1v2                                     : 0
      pg_vtt_0v6                                     : 0
      pg_vcc_1v1_cvl                                 : 0
      pg_3v3_1v8_cvl                                 : 0
      pg_vdd_0v8_cvl                                 : 0
      pg_avdd_eth_0v9_cvl                            : 0
      pg_avdd_pcie_0v9_cvl                           : 0
      pg_avddh_1v1_cvl                               : 0
      pg_pwr_qsfp0n                                  : 0
      pg_pwr_qsfp1n                                  : 0
      FPGA_THERM_SHDN                                : 0
      EDGE_PWR_SHDN                                  : 0
      FPGA_NCATTRIP                                  : 0
      VCC_12V_AUX_UV                                 : 0
      VCC_12V_PCIE_UV                                : 0
      VCC_3V3_PCIE_UV                                : 0
      VCC_12V_3V3_IN_OV                              : 0
      QSFPA_MODPRES                                  : 0
      QSFPB_MODPRES                                  : 0
    Power Good Record 1                              : 0x80000000
      pg_12v_aux_efuse                               : 0
      pg_12v_pcie_efuse                              : 0
      pg_vcc_5v                                      : 0
      pg_vcc_3v3                                     : 0
      pg_vccl_fpga_vid                               : 0
      pg_vccl_sdm_0v8                                : 0
      pg_vcch_0v9                                    : 0
      pg_vcch_gxer_1v1                               : 0
      pg_vcca_1v8                                    : 0
      pg_vccclk_gxer_2v5                             : 0
      pg_vpp_2v5                                     : 0
      pg_vccio_1v8                                   : 0
      pg_sdm_1v8                                     : 0
      pg_vcc_1v2                                     : 0
      pg_vtt_0v6                                     : 0
      pg_vcc_1v1_cvl                                 : 0
      pg_3v3_1v8_cvl                                 : 0
      pg_vdd_0v8_cvl                                 : 0
      pg_avdd_eth_0v9_cvl                            : 0
      pg_avdd_pcie_0v9_cvl                           : 0
      pg_avddh_1v1_cvl                               : 0
      pg_pwr_qsfp0n                                  : 0
      pg_pwr_qsfp1n                                  : 0
    Power Good Record 2                              : 0x00000000
      VCC_12V_AUX_UV                                 : 0
      VCC_12V_PCIE_UV                                : 0
      VCC_3V3_PCIE_UV                                : 0
      VCC_12V_3V3_IN_OV                              : 0
      FPGA_THERM_SHDN                                : 0
      EDGE_PWR_SHDN                                  : 0
      FPGA_NCATTRIP                                  : 0
      PM_ALERTN_3V3                                  : 0
      FPGA_VID_ALERTN                                : 0
      VR_VID_ALERTN                                  : 0
      FPGA_THERM_ALERTN                              : 0
      CVL_THERM_ALERTN                               : 0
      EDGE_PWR_WARN                                  : 0
      SI5392_LOL                                     : 0
    Sequencer Status 1                               : 0x80001000
      IDLE_ST                                        : 0
      PWR_CLS_DEC_FPGA_ST                            : 0
      PG_3V3_5V_ST                                   : 0
      WAIT_10MS_FPGA_ST                              : 0
      FPGA_GRP1_EN_ST                                : 0
      FPGA_GRP2_EN_ST                                : 0
      FPGA_GRP3_EN_ST                                : 0
      PG_VTT_0V6_CHK_ST                              : 0
      FPGA_GRP1_PWR_DWN_ST                           : 0
      FPGA_GRP2_PWR_DWN_ST                           : 0
      FPGA_GRP3_PWR_DWN_ST                           : 0
      FPGA_PWR_ON_ST                                 : 0
      FPGA_PWR_OFF_ST                                : 1
      PWR_CLS_DEC_CVL_ST                             : 0
      CVL_VCC_1V1_ST                                 : 0
      CVL_WAIT_100US_ST                              : 0
      CVL_3V3_1V8_EN_ST                              : 0
      CVL_VDD_0V8_EN_ST                              : 0
      CVL_AVDD_ETH_EN_ST                             : 0
      CVL_AVDD_PCIE_1V1_EN_ST                        : 0
      CVL_SI5392_CHK_ST                              : 0
      CVL_LAN_100US_WAIT_ST                          : 0
      CVL_LAN_PG_ST                                  : 0
      CVL_PWR_ON_ST                                  : 0
      CVL_LAN_PWR_DWN_ST                             : 0
      CVL_AVDD_PWR_DWN_ST                            : 0
      CVL_VDD_0V8_PWR_DWN_ST                         : 0
      CVL_3V3_1V8_PWR_DWN_ST                         : 0
      CVL_PWR_OFF_ST                                 : 1
    Sequencer Status 2                               : 0x00914000
      EN_VCCL_FPGA_VID                               : 0
      EN_VCCL_SDM_0V8_VCCH_0V9                       : 0
      EN_FPGA_GRP2                                   : 0
      EN_VPP_2V5                                     : 0
      EN_VCCIO_1V8_SDM_1V8                           : 0
      EN_VCC_1V2                                     : 0
      EN_3V3_CVL                                     : 0
      EN_1V8_CVL                                     : 0
      EN_VDD_0V8_CVL                                 : 0
      EN_AVDD_ETH_0V9_CVL                            : 0
      EN_AVDD_PCIE_0V9_CVL                           : 0
      EN_AVDDH_1V1_CVL                               : 0
      EN_PWR_QSFP0                                   : 0
      EN_PWR_QSFP1                                   : 0
      SI5392_LOL                                     : 1
      POWER_GOOD                                     : 0
      LAN_PWR_GOOD                                   : 0
      PM_ALERTN_3V3                                  : 0
      FPGA_VID_ALERTN                                : 1
      VR_VID_ALERTN                                  : 0
      FPGA_THERM_ALERTN                              : 0
      CVL_THERM_ALERTN                               : 1
      EDGE_PWR_WARN                                  : 0
    GPI Status                                       : 0x000000bc
      Max10 SEU error                                : 0
      FPGA SEU error                                 : 0
      QSFPA_MODPRES                                  : 1
      QSFPB_MODPRES                                  : 1
      PCIE_CLK_LOS                                   : 1
      PTP_CLK_LOL                                    : 1
      Si5392_LOL                                     : 0
      FPGA_nCATTrip                                  : 1
      STRAP_0_PCLASS0                                : 0
      STRAP_1_PCLASS1                                : 0
      SMBus_I2C_ADDR                                 : 0
      STRAP_3_CVL_PRSNT0_n                           : 0
      FLT_CFP_ISL                                    : 0
      IO_EXPNDR_INT_n                                : 0
    Sensor Failed                                    : 0x000013ff
      fpga_remote_temp                               : 1
      board_temp                                     : 1
      inlet_12v_pcie                                 : 1
      fpga_vcch_gxer                                 : 1
      max10_board_clk_pwr                            : 1
      cvl_core_vol_temp                              : 1
      board_3v3_vol_temp                             : 1
      fpga_vcch                                      : 1
      fpga_core_vol_pwr_temp                         : 1
      fpga_fab_tile_temp                             : 1
      overall_devices                                : 0
    Sensor Alert 1                                   : 0x00000000
    Sensor Alert 2                                   : 0x00000000
    Sensor Alert 3                                   : 0x00000000
  Sensor State Time                                  : Thu Jan  1 02:20:04 1970
    FPGA E-TILE Temperature#1                        :      0 °C
    FPGA E-TILE Temperature#2                        :      0 °C
    FPGA E-TILE Temperature#3                        :      0 °C
    FPGA E-TILE Temperature#4                        :      0 °C
    FPGA P-TILE Temperature                          :      0 °C
    FPGA FABRIC Max Temperature                      :      0 °C
    FPGA FABRIC DTS#1                                :      0 °C
    FPGA FABRIC DTS#2                                :      0 °C
    FPGA FABRIC DTS#3                                :      0 °C
    FPGA FABRIC DTS#4                                :      0 °C
    FPGA FABRIC DTS#5                                :      0 °C
    FPGA FABRIC RDTS#1                               :      0 °C
    FPGA FABRIC RDTS#2                               :      0 °C
    FPGA FABRIC RDTS#3                               :      0 °C
    FPGA FABRIC RDTS#4                               :      0 °C
    Board Top Near FPGA                              :     38 °C
    Board Bottom Near CVL                            :     37 °C
    Board Top East Near VRs Temperature              :     38 °C
    Columbiaville Die Temperature                    :     51 °C
    Board Rear Side Temperature                      :     38 °C
    Board Front Side Temperature                     :     35 °C
    QSFP1(Primary) Case Temperature                  :       N/A
    QSFP2(Secondary) Case Temperature                :       N/A
    FPGA Core Voltage Phase 0 VR Temperature         :     43 °C
    FPGA Core Voltage Phase 1 VR Temperature         :     42 °C
    FPGA Core Voltage Phase 2 VR Temperature         :     42 °C
    FPGA Core Voltage VR Controller Temperature      :     44 °C
    FPGA VCCH VR Temperature                         :     44 °C
    FPGA VCC_1V2 VR Temperature                      :     39 °C
    FPGA VCCH & VCC_1V2 VR Controller Temperature    :     42 °C
    3V3 VR Temperature                               :     47 °C
    CVL Core Voltage VR Temperature                  :     46 °C
    FPGA P-Tile Temperature [Remote]                 :     47 °C
    FPGA E-Tile Temperature [Remote]                 :     51 °C
    FPGA Core Temperature [Remote]                   :     71 °C
    FPGA Corner Temperature [Remote]                 :     70 °C
    Inlet 12V PCIe Rail [Voltage]                    :  11968 µV
    Inlet 12V PCIe Rail [Current]                    :   2337 µA
    Inlet 12V Aux Rail [Voltage]                     :  12040 µV
    Inlet 12V Aux Rail [Current]                     :   1154 µA
    Inlet 3V3 PCIe Rail [Voltage]                    :   3296 µV
    Inlet 3V3 PCIe Rail [Current]                    :   1466 µA
    Board Power                                      :  46695 µW
    FPGA Core Voltage Rail [Voltage]                 :    810 µV
    FPGA Core Voltage Rail [Current]                 :   5359 µA
    FPGA VCCH Rail [Voltage]                         :      1 µV
    FPGA VCCH Rail [Current]                         :      0 µA
    FPGA VCC_1V2 Rail [Voltage]                      :    485 µV
    FPGA VCC_1V2 Rail [Current]                      :      0 µA
    FPGA VCCH_GXER_1V1 & VCCA_1V8 [Voltage]          :   3344 µV
    FPGA VCCH_GXER_1V1 & VCCA_1V8 [Current]          :   1133 µA
    FPGA VCCIO_1V2 [Voltage]                         :   1192 µV
    FPGA VCCIO_1V2 [Current]                         :   2840 µA
    CVL Non Core Rails Inlet [Voltage]               :   3344 µV
    CVL Non Core Rails Inlet [Current]               :    600 µA
    MAX10 & Board CLK PWR 3V3 Inlet [Voltage]        :   3288 µV
    MAX10 & Board CLK PWR 3V3 Inlet [Current]        :   1386 µA
    CVL Core Voltage Rail [Voltage]                  :     35 µV
    CVL Core Voltage Rail [Current]                  :     62 µA
    Board 3V3 VR [Voltage]                           :   3363 µV
    Board 3V3 VR [Current]                           :    750 µA
    QSFP 3V3 Rail [Voltage]                          :   3344 µV
    QSFP 3V3 Rail [Current]                          :     80 µA
    QSFP (Primary) Supply Voltage Rail [Voltage]     :       N/A
    QSFP (Secondary) Supply Voltage Rail [Voltage]   :       N/A
    VCCCLK_GXER_2V5 Voltage                          :    125 µV
    AVDDH_1V1_CVL Voltage                            :      0 µV
    VDDH_1V8_CVL Voltage                             :    571 µV
    VCCA_PLL Voltage                                 :    491 µV
    VCCRT_GXER_0V9 Voltage                           :      0 µV
    VCCRT_GXPL_0V9 Voltage                           :      0 µV
    VCCH_GXPL_1V8 Voltage                            :    968 µV
    VCCPT_1V8 Voltage                                :    969 µV
    VCC_3V3_M10  Voltage                             :   3334 µV
    VCC_1V8_M10  Voltage                             :   1805 µV
    VCC_1V2_EMIF1_2_3 Voltage                        :    490 µV
    VCC_1V2_EMIF4_5 Voltage                          :    493 µV
    VCCA_1V8  Voltage                                :    970 µV
    VCCH_GXER_1V1 Voltage                            :     90 µV
    AVDD_ETH_0V9_CVL  Voltage                        :      0 µV
    AVDD_PCIE_0V9_CVL Voltage                        :      3 µV
    Virt FPGA Temperature                            :      0 °C
  Sensor Status Time                                 : Thu Jan  1 02:20:04 1970
    INA3221 1 Mask Enable                            : 0x00000003
    INA3221 2 Mask Enable                            : 0x00000003
    INA3221 3 Mask Enable                            : 0x00000001
    IR38062 Status Word                              : 0x00000840
    IR38062 Voltage                                  :       0 µV
    IR38062 Current                                  :       0 µA
    IR38062 Temperatur                               :       0 °C
    IR38062 Input                                    :       0 µV
    IR38062 Status CML                               : 0x00000000
    IR38063 Status Word                              : 0x00000000
    IR38063 Voltage                                  :       0 µV
    IR38063 Current                                  :       0 µA
    IR38063 Temperatur                               :       0 °C
    IR38063 Input                                    :       0 µV
    IR38063 Status CML                               : 0x00000000
    ISL68220 Status Word                             : 0x00000840
    ISL68220 Voltage                                 :       0 µV
    ISL68220 Current                                 :       0 µA
    ISL68220 Temperatur                              :       0 °C
    ISL68220 Input                                   :       0 µV
    ISL68220 Status CML                              : 0x00000000
    ED8401 Status                                    : 0x00000000
  PCI Error Status Time                              : Thu Jan  1 02:20:14 1970
```

tswhison commented on tools/fpgainfo/events.c:
```diff
@@ -0,0 +1,124 @@
+
+		switch (opt) {
+		case 'b':
+			first = first + strtoul(optarg, NULL, 0);
+			last = last + strtoul(optarg, NULL, 0);
+			break;
+		case 'c':
+			last = last + strtoul(optarg, NULL, 0);
```
Need some safeguards here in case optarg is not a number?

hundeboll-silicom replied:
I added a helper function to check the conversion.

tswhison reviewed (commented):
Could you please add unit tests for the new fpgainfo and libboard functionality?

tswhison reviewed (changes requested):
Please add unit tests as requested.

hundeboll-silicom commented:
I updated the branch with unittests, but they depend on https://github.com/OPAE/opae-test/pull/31

Also added some more parsing of the log bits itself, and fixed the lookup of sensor descriptions.

hundeboll-silicom commented:
@michelleyho @r-rojo Please review

tswhison commented on opae-libs/cmake/config/config.h.in:
```diff
@@ -231,6 +231,7 @@
 #define OPAE_ASE_CFG_INST_PATH "@CMAKE_INSTALL_PREFIX@/share/opae/ase/opae_ase.cfg"
 
 #define OPAE_MODULE_SEARCH_PATHS        \
+  "@LIBRARY_OUTPUT_PATH@/",             \
```
We don't want this for the production case, as it violates our security standards.  Use LD_LIBRARY_PATH instead.

rchriste1 replied:
I will revert it. Thanks.

rchriste1 reviewed (approved):
Branch rebased to current master. Smaller fixes like "mA", "mV", -c counts corrected, and changing terminology Pass/Fail to Yes/no, all squashed into corresponding original commits.
